### ISSUE TITLE
Radiation property library

### ISSUE THREAD

**Issue by BYUignite** [Cantera/enhancements#72]
**Abstract**

A brief description of the work being done.

We have developed a radiation property library called radlib (https://github.com/BYUignite/radlib) the implements the Planck Mean (PM), WSGG, and the new RCSLW models. These models provide absorption coefficients and weighting factors that can be combined with a solver for the radiative transport equation. The code is fully documented, verified, and includes several examples. The library is written in C++ and includes a Python interface written in Cython. Each of the three models currently implemented includes CO2, H2O, and soot, with RCSLW including CO, and PM including CO and CH4. One of many advantages of Cantera is that it provides a library of thermochemical and reaction properties that simplify the development of user tools. We would like to include additional radiation properties in Cantera for use in user tools.

**Motivation**

Describe the need for the work being done:

* What problem is it trying to solve?
Provide radiation properties for a varying levels of accuracy, computational cost, and complexity.

* Who is affected by the change?
These models will aid researchers who want to implement radiative transport solvers by providing models for the radiative properties in participating media. Examples include CFD solvers, and laminar flames, among others.

* Why is this a good solution?

The models noted seem consistent with the purpose and approach of Cantera. The radlib code is only a starting point and would require appropriate integration into Cantera, which we are willing to do, with some guidance. The Planck Mean model is already implemented in Cantera. We would like to extend the capabilities to other models.

**Description**
Please see https://github.com/BYUignite/radlib for a detailed description, including code documentation: https://ignite.byu.edu/radlib_documentation/. Code examples and plots are shown here: https://github.com/BYUignite/radlib/blob/master/examples/python/run_and_plot_examples.ipynb

**Alternatives**

If any alternative solutions to solving the same problem have been considered, describe them here, and explain why the chosen approach is preferred.

**References**

Links to a development branch in your fork of the Cantera repository, Pull Requests, GitHub Issues, Users' Group topics, or other relevant material.

https://github.com/BYUignite/radlib

**Comment by bryanwweber** [comment]
Hi @BYUignite! Thank you so much for posting this. I don't have any updates other than to say I'm looking forward to seeing how this can be integrated with Cantera!

**Comment by BYUignite** [comment]
Hi, thanks Brian, I’m glad this is of interest. I think we last met at the Cantera workshop at the US Combustion Meeting at CalTech. I was the program chair for that meeting, so I was a bit distracted with other things at the time.

I’m a huge fan of Cantera and use it regularly, but I’ve not contributed directly to the code. I would like to do that in ways that would be useful (and always subject to me thinking I have more time than I do :-( I worked with Harry Moffett when he was developing CADS, and I’ve done a lot of work with John Hewson. We are also developing a number of models for representing soot (or other aerosol) size distributions, including sectional, MOMIC, QMOM, DQMOM, and others, and these might be of interest as well.

Perhaps the thing to do with the radiation is to issue a pull request? I can do that, and read the contribution guides, but it’s likely that I’ll run into a couple questions that might be better to have answered before implementing, issuing the request, and then fixing. For example, the RCSLW radiation model reads data from a file and I’d want to put those files in the right place. (They are not configuration files, and the data could be hard-coded, but they are a couple MB, so that’s not ideal.) Similarly, I’d want to make sure the Python (Cython) interface was done right. Is the Cantera group the best place to communicate, or email, (or even phone)?

Best Regards,

David

David O. Lignell
Professor, Chemical Engineering
Brigham Young University
801-422-1772 | http://ignite.byu.edu

On Jan 12, 2021, at 6:37 PM, Bryan W. Weber <notifications@github.com<mailto:notifications@github.com>> wrote:



Hi @BYUignite<https://github.com/BYUignite>! Thank you so much for posting this. I don't have any updates other than to say I'm looking forward to seeing how this can be integrated with Cantera!

**Comment by ischoegl** [comment]
I am posting a link to a current PR Cantera/cantera/pull/965 here that I believe has overlap. I haven’t looked into radiation properties myself at this point.

**Comment by BYUignite** [comment]
Thanks for sharing. This is interesting.

David O. Lignell
Professor, Chemical Engineering
Brigham Young University
801-422-1772 | http://ignite.byu.edu

On Feb 9, 2021, at 7:28 AM, Ingmar Schoegl <notifications@github.com<mailto:notifications@github.com>> wrote:



I am posting a link to a current PR Cantera/cantera/pull/965<https://github.com/Cantera/cantera/pull/965> here that I believe has overlap. I haven’t looked into radiation properties myself at this point.

**Comment by lavrenyukiv** [comment]
You use the same polynomials for calculating the Planck-averaged coefficients as https://www.sandia.gov/TNF/radiation.html. It is not clear how accurate these polynomials are at high temperatures (small approximation errors lead to huge deviations in energy), so using linear interpolation in logarithm scale between "exact" values is preferably. Also polynomial for CO2 leads to negative absorption coefficient at low temperatures.

**Comment by BYUignite** [comment]
Changing the Planck-mean coefficients would be easy to do, and we could guard properly against value ranges.

**Comment by wandadars** [comment]
@BYUignite  Am I correct in my understanding that the radiation properties are totally uncoupled to the radiation solver that someone chooses to use?   If RadLib has those 3 models for the radiation coefficients, any of those three could be used in any other models that compute heat transfer? Of is it a more complex relation where property-method A works for solver-methods 1 and 2, but not 3, etc.?

**Comment by BYUignite** [comment]
Yes, the Planck Mean polynomial coefficients could be improved. If other correlations are available we could use them easily. They could also be generated from the data available in the RCSLW model, among others.

On Feb 9, 2021, at 12:56 PM, lavrenyukiv ***@***.***> wrote:



You use the same polynomials for calculating the Planck-averaged coefficients as https://www.sandia.gov/TNF/radiation.html. It is not clear how accurate these polynomials are at high temperatures (small approximation errors lead to huge deviations in energy), so using linear interpolation in logarithm scale between "exact" values is preferably. Also polynomial for CO2 leads to negative absorption coefficient at low temperatures.

**Comment by BYUignite** [comment]
That’s correct. The solver is completely separate from the radiation properties. In RadLib, we provide a 1-D ray tracing solver that couples to RadLib for illustration of how the model can be used. Any of the three models in RadLib can be used with an existing solver. We implemented RadLib into NIST’s Fire Dynamics Simulator (FDS) in that way. FDS had a discrete ordinates solver that could include a WSGG model. We used RadLib to provide the gray gas absorption coefficients and weights. The implementation was trivial. I don’t know that I would say RadLib would work with any solver for the radiative transport equation. It’s really meant for solvers that make use of gray gases of some kind, since the three methods in RadLib are of that category. But that is probably the most relevant space for practical applications. The models also differ: Plank Mean only considers a single gas (non-spectral), and so is ideal for solvers that assume a single gas. Other methods we’ve considered implementing for a single gas include those based on emissivity correlations with a mean beam length that relates the emissivity to the absorption coefficient. For solvers that hard-code a given number of gray gases, the RCSLW model would make the most sense since it allows variable number of gases. Conversely, Bordbar’s WSGG model that RadLib implements assumes a fixed number of gases.

David O. Lignell
Professor, Chemical Engineering
Brigham Young University
801-422-1772 | http://ignite.byu.edu

On Jan 10, 2025, at 2:28 PM, Chris Neal ***@***.***> wrote:



@BYUignite<https://github.com/BYUignite> Am I correct in my understanding that the radiation properties are totally uncoupled to the radiation solver that someone chooses to use? If RadLib has those 3 models for the radiation coefficients, any of those three could be used in any other models that compute heat transfer? Of is it a more complex relation where property-method A works for solver-methods 1 and 2, but not 3, etc.?
